### PR TITLE
refactor(server): job discovery

### PR DESCRIPTION
**Pull request by jrasm91**
Auto-register/bind job handlers to jobs with a new decorator `@OnJob({ name: JobName, queue: QueueName })`. This replaces the previous `JOBS_TO_QUEUE` object and microservices.service.ts implementation. I also found and fixed a few discrepancies with the types used in the handlers (mismatched the `JobItem` type).

### CODE REVIEW

**Review by danieldietzler — APPROVED** [review]
Awesome, great job! :)

File: server/src/services/media.service.ts [hunk 1]
```diff
@@ -39,7 +39,8 @@ export class MediaService extends BaseService {
   private maliOpenCL?: boolean;
   private devices?: string[];
 
-  async handleQueueGenerateThumbnails({ force }: IBaseJob): Promise<JobStatus> {
+  @OnJob({ name: JobName.QUEUE_GENERATE_THUMBNAILS, queue: QueueName.THUMBNAIL_GENERATION })
```
Comment (alextran1502) [inline server/src/services/media.service.ts:42]: What happens if I put this decorator on two different methods in this Service?

Comment (jrasm91) [reply server/src/services/media.service.ts:42]: It throws an error and fails to start

**Review by alextran1502 — APPROVED** [review]
Amazing refactor work here! I wonder if this is a point where we can write some documents on the flow of job creation since there are many parts involved.

**Review by zackpollard — APPROVED** [review]

**Pull request merged**